islandora_xacml_editor.drush.inc

```
@@ -0,0 +1,184 @@
+<?php
+
+/**
+ * @file
+ * Drush hooks.
+ */
+
+/**
+ * Implements hook_drush_command().
+ */
+function islandora_xacml_editor_drush_command() {
+  $items = array();
+
+  $items['islandora_xacml_editor_apply_policy'] = array(
+    'aliases' => array('ixeap'),
+    'description' => dt('Apply XACML policy to target object.'),
+    'bootstrap' => DRUSH_BOOTSTRAP_DRUPAL_LOGIN,
+    'examples' => array(
+      'Apply policy.xml to \'islandora:57\', use traversal to target child objects.' => 'drush -v --user=1 islandora_xacml_editor_apply_policy --policy=/tmp/policy.xml --pid=islandora:57 --traversal',
+    ),
+    'options' => array(
+      'policy' => array(
+        'description' => dt('The path to an XML file containing the XACML policy configuration to be applied.'),
```

This... seems like we could open a new set of issues, if people attempt to write XACML policies by hand, and then push them onto objects... our XACML editor's behaviour would then be undefined.

---

Could do some basic validation, instantiating a [`XACML` object](https://github.com/Islandora/islandora_xacml_editor/blob/ed2cb7b5ad036470afa2e763d12c13849c611568/api/includes/xacml.inc#L741-L884), and passing it the read file... it's somewhat naive, but it [will attempt to reject XACML that was not generated with the editor](https://github.com/Islandora/islandora_xacml_editor/blob/ed2cb7b5ad036470afa2e763d12c13849c611568/api/includes/parser.inc#L47-L50) (and some other validation besides).

---

I... guess since this ultimately goes through [the `IslandoraUpdatePolicy` class](https://github.com/Islandora/islandora_xacml_editor/blob/fe44f688997ee0bc9e04145bbf9c0f88a51348c9/includes/batch.inc#L76-L127), which eventually [uses the `IslandoraXacml` class](https://github.com/Islandora/islandora_xacml_editor/blob/fe44f688997ee0bc9e04145bbf9c0f88a51348c9/includes/batch.inc#L119-L126) (which is a subclass of `Xacml`), and so should similarly fail to parse if passed something invalid...

... May want to mention here in the description that we expect this policy to have been generated by the XACML editor, at least?